units/schedule_patch.go

```diff
@@ -0,0 +1,73 @@
+package units
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/pkg/errors"
+)
+
+// SchedulePatch schedules a patch. It returns an error and an HTTP status code.
+func SchedulePatch(ctx context.Context, patchId string, version *model.Version, patchUpdateReq model.PatchUpdate) (int, error) {
```

Comment:
Tried to move a lot of this logic into model, but overall it couldn't be moved because of a dependency cycle with units. Not sure if this should exist in rest/data instead.